.github/workflows/fetch-dev-artifacts.yaml

```diff
@@ -0,0 +1,97 @@
+name: Monthly Artifact Bump
+
+permissions:
+  contents: write
+  pull-requests: write
+
+on:
+  schedule:
+    - cron: '10 4 1-7 * 5'
+  workflow_dispatch:
+  
+
+jobs:
+  fetch-artifacts:
+   runs-on: ubuntu-latest
+   steps:
+    - name: Checkout Repository
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+    - name: Set Up Git User
+      run: |
+          git config --global user.name "Dev Artifact Bot"
+          git config --global user.email "github-actions[bot]@users.noreply.github.com"
+   
+    - name: Configure AWS CLI
+      uses: aws-actions/configure-aws-credentials@v2
+      with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_USERNAME }}
+          aws-secret-access-key: ${{ secrets.AWS_ACCESS_PASSWORD }}
+          aws-region: eu-west-1
+    - name: Configure CodeArtifact Authentication Token 
+      run: |
+          CODEARTIFACT_TOKEN=`aws codeartifact get-authorization-token --domain build-service-live --domain-owner ${{ secrets.AWS_ACCOUNT_ID }} --query authorizationToken --output text`
+          echo "::add-mask::$CODEARTIFACT_TOKEN"
+          echo "CODEARTIFACT_TOKEN=$CODEARTIFACT_TOKEN" >> "$GITHUB_ENV"
+
+    - uses: actions/checkout@v3
+    
+    - name: Determine latest neo4j CI version
+      run: |
+          neo4j_version_base=$(cat neoBaseVersion.txt)
+          echo "neo4j_version_base=$neo4j_version_base" >> $GITHUB_ENV
+          echo "Found neo4j_version_base=$neo4j_version_base"
+
+    - name: fetch parser artifacts from latest neo4j CI
+      run: |
+          aws --no-cli-pager codeartifact get-package-version-asset --domain build-service-live --repository ci-live --format maven --namespace org.neo4j --package cypher-v25-antlr-parser --package-version $neo4j_version_base-SNAPSHOT --asset $(aws --no-cli-pager codeartifact list-package-version-assets --domain build-service-live --repository ci-live --format maven --namespace org.neo4j --package cypher-v25-antlr-parser --package-version $neo4j_version_base-SNAPSHOT --query 'assets[?ends_with(name, `grammar.zip`)].name | [0]' --output text --domain-owner ${{ secrets.AWS_ACCOUNT_ID }}) grammar.zip --domain-owner ${{ secrets.AWS_ACCOUNT_ID }}
```

Comment:
It would be good to extract the asset part:

```
$(aws --no-cli-pager codeartifact list-package-version-assets --domain build-service-live --repository ci-live --format maven --namespace org.neo4j --package cypher-v25-antlr-parser --package-version $neo4j_version_base-SNAPSHOT --query 'assets[?ends_with(name, `grammar.zip`)].name | [0]' --output text --domain-owner ${{ secrets.AWS_ACCOUNT_ID }})
```

to a different variable?

Reply:
`--domain-owner` is repeated twice here